reason_4.0/lib/core/classes/social/twitter.php

```diff
@@ -117,7 +117,7 @@ function social_account_pre_show_form($cm)
 	function social_account_run_error_checks($cm)
 	{
 		$account_id = $cm->get_value('account_id');
-		if ( !check_against_regexp($account_id, array('/^[a-z\d._]*$/i')) )
+		if ( !checkagainst_regexp($account_id, array('naturalnumber')) && !check_against_regexp($account_id, array('/^[a-z\d.]*$/i')) )
```

Comment:
This looks like a better approach: `preg_match('/^[A-Za-z0-9_]{1,15}$/', $username)`

http://stackoverflow.com/questions/4424179/how-to-validate-a-twitter-username-using-regex